src/views/dashboard/components/core/AppBar.vue

```diff
@@ -200,63 +146,147 @@
       },
     },
     data: () => ({
-      modalList: [
+      modalList: [],
+      LoginFieldsList: [
         { name: 'user', label: 'Username', default: '' },
-        { name: 'pwd', label: 'Password', default: '' },
+        { name: 'pwd', label: 'Password', default: '', type: 'password' },
       ],
-      notifications: [
-        'Mike John Responded to your email',
-        'You have 5 new tasks',
-        'You\'re now friends with Andrew',
-        'Another Notification',
-        'Another one',
+      SigninFieldsList: [
+        { name: 'user', label: 'Username', default: '' },
+        { name: 'pwd', label: 'Password', default: '', type: 'password' },
+        { name: 'name', label: 'Name', default: '' },
       ],
       menuUserItems: [
         { title: 'Log In', id: '1' },
         { title: 'Sign Up', id: '2' },
-        { title: 'Test', id: '3' },
       ],
-      showUserModalS: false,
-      showUserModalL: false,
+      menuLogoutItem: [
+        { title: 'log out', id: '3' },
+      ],
+      menuItemList: [],
+      snack: {
+        show: false,
+        text: '',
+        color: 'red',
+      },
+      timeout: 2000,
       showEditModal: false,
+      showMenu: false,
+      modalFunction: '',
+      modalTitle: '',
+      modalButtonText: '',
+      accountIconColor: '',
     }),
     computed: {
       ...mapState(['drawer']),
     },
+    mounted () {
+      if ('token' in localStorage) {
+        this.accountIconColor = 'success'
+      } else {
+        this.accountIconColor = ''
+      }
+    },
     methods: {
       ...mapMutations({
         setDrawer: 'SET_DRAWER',
         setUserInfo: 'SET_USER',
       }),
+      logout () {
+        localStorage.removeItem('token')
+        this.snack = { show: true, text: 'You are logged out!', color: 'green' }
+        this.accountIconColor = ''
+      },
+      menuItems () {
+        if ('token' in localStorage) {
+          this.menuItemList = this.menuLogoutItem
+        } else {
+          this.menuItemList = this.menuUserItems
+        }
+      },
       showModal (bmenu) {
         if (bmenu === '1') {
-          this.showUserModalL = true
+          this.modalList = this.LoginFieldsList
+          this.modalFunction = this.loginSubmit
+          this.modalTitle = 'Introduce your credentials to log in!'
+          this.modalButtonText = 'Log in'
+          this.showEditModal = true
         } else if (bmenu === '2') {
-          this.showUserModalS = true
-        } else if (bmenu === '3') {
+          this.modalList = this.SigninFieldsList
+          this.modalFunction = this.signinSubmit
+          this.modalTitle = 'Introduce your credentials to sign up!'
+          this.modalButtonText = 'Sign up'
           this.showEditModal = true
+        } else if (bmenu === '3') {
+          this.logout()
+          this.showEditModal = false
         }
```

Comment:
this is very poorly done actually.
Some pointers to make it prettier: 

1. Store all the modal information in one object:

```
data () => ({
//...
modalInfo = {fields: [], function: {}, title: '', buttonText: ''}
//..
})
```

2. Then, create a list of three objects inside data:


```
data () => ({
//...
modalInfo = {fields: [], function: {}, title: '', buttonText: ''}
listOptions = [ {fields: [], function: {}, title: '', buttonText: '', menuTitle: "Log in"}, { }, { }  ]
//..
})
```

3. In the html, you iterate over the list called `listOptions` and you pass the complete object to the function `showModal()`.
4. The function `showModal(object)` does:

```
this.modalInfo = object
```

5. When calling the `testModal` you just use the `modalInfo` object:

```
   <test-modal
      v-model="showEditModal"
      :fields="modalInfo.fields"
      :title="modalInfo.title"
      :buttonText="modalInfo.buttonText"
      @submit-form="modalInfo.function"
    />
```

You can later automatize this last part to only pass one object. Look [here](https://vuejs.org/v2/guide/components-props.html#Passing-the-Properties-of-an-Object) and [here](https://vuejs.org/v2/guide/components.html#A-Single-Root-Element)